src/main/java/org/apache/flink/state/benchmark/ListStateBenchmarkBase.java

```diff
@@ -79,19 +87,37 @@ public void tearDown() {
 
     @Benchmark
     public void test1Update() throws Exception {
-        keyedStateBackend.setCurrentKey(random.nextLong(setupKeyCount));
-        listState.update(Collections.singletonList(random.nextLong()));
+        int currentIndex = getCurrentIndex();
```

Comment:
There is actually a better way to this test initialisation, please check this:

http://tutorials.jenkov.com/java-performance/jmh.html#state-setup-and-teardown

the idea is that you should do the time consuming setup in in context, once per `Level.Trial` (the default), `Level.Iteration` or `Level.Invocation`.

Secondary concern is when you are passing input parameters not through the context, the code is subject to constant foldings/dead code elimination etc: http://tutorials.jenkov.com/java-performance/jmh.html#writing-good-benchmarks

To avoid those kind of issues, a good benchmark should get it's input data through JMH's `@State` and should return something from the benchmark methods. (`void` in `public void test1Update` is also dangerous)

(the same applies to other benchmarks)

Reply:
Thanks for the reference of `State` object setUp/tearDown annotation, will init the inputs of benchmark methods following this way.

Also checked the reference to prevent constant folding/dead code and confirmed current codes are in good shape (already using `BlackHole` or having `return` statement for getter benchmark, and all benchmark with no return value have specific operations such as `HeapListState.update`)